Beta NWT_CLM Scripts

hhollandmoritz opened the pull request:
@wwieder a repository for your perusal. 

There are 5 scripts, a set of CLM instructions, and a Readme file for the user describing all of the scripts and the workflow. 

The most complete and well-thought out files are `CLM_instructions.md`, `flow.lter.clm.R`, and `supplemental_cleaning.R`. The post-processing scripts are newer and I have had less time to troubleshoot them. As a result, they don't have as many bells and whistles and probably are more buggy than the pre-processsing scripts.

wwieder commented:
I may have more reviews, but ended up just changing things in the PRs I've made

hhollandmoritz commented on CLM_instructions.md:
````diff
@@ -109,10 +364,10 @@ Modify the *.xml files to specify the run conditions and computer-use requiremen
 ```bash
 # Starting conditions
-./xmlchange --file env_run.xml  --id CLM_FORCE_COLDSTART --val on # CLM forced to do a cold start with arbitrary initial conditions (also used for spin-up). finidat is set to blanks
+./xmlchange --file env_run.xml  --id CLM_FORCE_COLDSTART --val on # CLM forced to do a cold start with arbitrary initial conditions (also used for spin-up). 
````
Hmm, maybe this could be fixed with a separate file of xml change commands that don't have comments? That way the context is there for people who want it, but people can copy and paste from the file? Or, maybe I could add a section to the end of the document that is just the xmlchange commands without comments, and then provide a link to that section. That would keep all the information in one file.

hhollandmoritz commented on CLM_instructions.md:
````diff
@@ -1,19 +1,237 @@
+```bash
+module load nco # load the nco command module to access ncap2 program
+# Change the percent sand and clay variables
+ncap2 -s PCT_SAND=PCT_SAND*0+39;PCT_CLAY=PCT_CLAY*0+23 /path/to/your/surfacedata/surfdata_0.9x1.25_16pfts_CMIP6_simyr1850_254.42_40.05_c170706.nc /path/to/your/surfacedata/surfdata_0.9x1.25_16pfts_CMIP6_simyr1850_254.42_40.05_c170706_pctsandclay.nc
````
Interesting, I didn't have any issue with this, I wonder what's different between our accounts? Either way, I'll add the quotes in.

hhollandmoritz replied:
Done!

hhollandmoritz commented on CLM_instructions.md:
````diff
@@ -1,19 +1,237 @@
+```bash
+module load python
+python singlept
+```
````
Oops! You're totally right. Sorry about forgetting that.

hhollandmoritz commented on README.md:
```diff
@@ -1,11 +1,245 @@
+
+### 1. `tvan_supplemental_cleaning.R`
+
+This script cleans up Tvan L1 data that has been produced with the Niwot LTER `tvan_L1_preprocess.R` script from the Niwot LTER's repository. It is a temporary script meant to add a few extra cleaning steps to the L1 tvan data, until that data can be uploaded to AmeriFlux. The script reads in ReddyProc-ready data output by the `tvan_L1_preprocess.R` script, filters several problem spots, plots ye ...
```
Sounds good!

hhollandmoritz replied:
The script now reflects this change.

hhollandmoritz commented on tvan_supplemental_cleaning.R:
```diff
@@ -722,7 +724,7 @@ if (makeplots) {
 # Download saddle grid snow_depth_data
 message(paste0("Downloading Saddle Met data, please cite: \n",
-               "CITATION (Accessed ",Sys.Date(), ")"))
+               "CITATION NEEDED (Accessed ",Sys.Date(), ")"))
```
Good idea.

hhollandmoritz commented on tvan_supplemental_cleaning.R:
```diff
@@ -722,7 +724,7 @@ if (makeplots) {
 ##############################################################################
 # Download saddle grid snow_depth_data
 message(paste0("Downloading Saddle Met data, please cite: \n",
```
Yep, perils of copying code hastily.

hhollandmoritz commented on tvan_supplemental_cleaning.R:
```diff
@@ -722,7 +724,7 @@ if (makeplots) {
 ##############################################################################
 # Download saddle grid snow_depth_data
```
Ack, I thought I'd fixed all of those. I'll have to go through it again with a fine-toothed comb and make sure.

hhollandmoritz replied:
Fixed this.

hhollandmoritz commented on tvan_supplemental_cleaning.R:
```diff
@@ -43,9 +43,11 @@ options(stringsAsFactors = F)
 makeplots <- FALSE# TRUE
```
Will do.

hhollandmoritz commented on tvan_supplemental_cleaning.R:
```diff
@@ -43,9 +43,11 @@ options(stringsAsFactors = F)
+# The output directory for the script. It is recommended but not required that this
+# be the same directory that holds the Reddyproc-ready files produced by 
+# tvan_L1_preprocess.R
+DirOutBase <- "~/Downloads/Tvan_out/Reddy_proc_readyData"
```
Thanks!

wwieder commented on flow.obs.R:
```diff
@@ -43,11 +43,10 @@ ver <- paste0("v",format(Sys.time(), "%Y%m%dT%H%m"))
 DirOutBase <- paste0("~/Downloads/OBS/data",ver)
```
Is `ver` hardcoded elsewhere in the code?  I keep getting errors, but I'm not sure it's because I'm not putting results in dowloads or because of the date we're using?

hhollandmoritz replied:
I've removed `ver` from all of these scripts. It was not particularly useful given how rarely the data change and because it was timestamped it was more of a hassle than it was worth.

hhollandmoritz commented on README.md:
```diff
@@ -1,11 +1,245 @@
+
+*This script will be rendered obsolete once the Tvan data is available on AmeriFlux 
+
+![NWT_CLM workflow overview](./images/Conceptual_diagram.png)
```
I've added the .svg version of the figure to the repository. But have not made changes 2 or 3 yet.

hhollandmoritz commented on flow.sim.R:
```diff
@@ -26,15 +26,17 @@ lapply(packReq, function(x) {
 
 # The name of the netcdf file from the simulation you want to work with
 ncdf_fp <- "2000datm_CLM50bgc_nwt_DM.clm2.h0.2008-01-01-00000.nc"
+ncdf_fp <- "test.nc"
```
I think most of the unused code is gone now. Let me know if you still see areas of it.

hhollandmoritz commented on flow.sim.R:
```diff
@@ -26,15 +26,17 @@ lapply(packReq, function(x) {
+veg_com <- "DM" # Options: "FF", "DM", "WM", "MM", "SB", NA
 
 ##############################################################################
 # Static workflow parameters - these are unlikely to change
```
I removed nsteps. It doesn't serve any purpose since the script dynamically determines the number of timesteps.

hhollandmoritz commented:
@wwieder Lots of changes here. The biggest and most important is that I implemented the ability to run multiple vegetation types through flow.sim.R at once. I also fixed some of the smaller line-by-line comments you had from earlier, and added diel obs-vs-sim scatter plots to the comparison plotting. 

I tried moving the timestamp backwards by one in the simulation data, it does look better but not by much. 

Finally, I fixed some of the aesthetics of the plots where panels were in weird order.

Oh, and there's also snow_depth plotting now. (I can't remember if that was there before or not), but it's there and because you can run multiple vegetation communities through flow.sim.R, you can now compare snow depths across veg. communities in the comparison plotting script.

hhollandmoritz commented:
I just added a "script status" section for each script in the workflow. Hopefully this will make it more easy to identify what needs to be done to improve the workflow going forward. (Though it's no substitute for the "Issues" tab on github).

wwieder commented:
merge beta back into main
